testng-core/src/main/java/org/testng/reporters/MethodInvocationKey.java

```diff
@@ -0,0 +1,48 @@
+package org.testng.reporters;
+
+import java.util.Arrays;
+import java.util.Objects;
+import org.testng.ITestNGMethod;
+
+class MethodInvocationKey {
+  private final Object methodIdentity;
+  private final Object[] parameters;
+  private final int currentInvocationCount;
+
+  MethodInvocationKey(
+      ITestNGMethod methodIdentity, Object[] parameters, int currentInvocationCount) {
```

Comment:
According of the usage, methodIdentity should be typed Object.
But please, use T instead of Object everywhere you can

Reply:
I chose to restrict it to being `ITestNGMethod` since it is now used only internally and there are no external usages.